wikimedia/wikipedia-ios
# Diffs: Update cell accessibility labels

**Phabricator:** https://phabricator.wikimedia.org/T343582 (partially)

### Notes
This PR improves the accessibility of the diff view with targeted improvements when VoiceOver is enabled.

Changes include:
- Adds an "accessibility text" element per diff view model to customize the VoiceOver experience
- Changes to the diff change cell labels to "source code" context to read as many characters as possible
- Updates the VoiceOver experience for addition/removal of content (in order to communicate the red/green deletion and insertion styling)
- Updates contextual cell lines to read "Contextual line, unchanged"
- Updates line/section moving changes to indicate distance from current element
- Updates the collapse/expand buttons to indicate line numbers they will collapse/expand

It would have been nice to leverage some of the attributed string [accessibility keys](https://developer.apple.com/documentation/foundation/nsattributedstring/key) to directly apply accessibility traits on the constructed attributed string without having to construct separate strings for VoiceOver purposes.
But unfortunately, [accessibilityTextCustom](https://developer.apple.com/documentation/foundation/nsattributedstring/key/2865778-accessibilitytextcustom) doesn't appear to work as expected (or I'm misinterpreting its intent).


### Test Steps

I'd recommend we rely on QA for extensive testing, but you can lightly check the changes above in any article diff view: 

1. Open an Article
2. Scroll to article history
3. Confirm VoiceOver is on, then tap a change
4. Confirm the contained diff change area can be navigated via VoiceOver
5. Confirm VoiceOver changes from above (if present) in the particular diff you're on
